packages/kbn-docs-utils/src/api_docs/build_api_docs_cli.ts

```diff
@@ -55,36 +77,118 @@ export function runBuildApiDocsCli() {
         pluginApiMap[plugin.manifest.id] = getPluginApi(project, plugin, plugins, log);
       });
 
-      const missingApiItems: { [key: string]: string[] } = {};
+      // Mapping of plugin id to the missing source API id to all the plugin API items that referenced this item.
+      const missingApiItems: { [key: string]: { [key: string]: string[] } } = {};
 
       plugins.forEach((plugin) => {
         const id = plugin.manifest.id;
         const pluginApi = pluginApiMap[id];
-        removeBrokenLinks(pluginApi, missingApiItems, pluginApiMap);
+        removeBrokenLinks(pluginApi, missingApiItems, pluginApiMap, log);
       });
 
+      const reporter = CiStatsReporter.fromEnv(log);
       plugins.forEach((plugin) => {
+        // Note that the filtering is done here, and not above because the entire public plugin API has to
+        // be parsed in order to correctly determine reference links, and ensure that `removeBrokenLinks`
+        // doesn't remove more links than necessary.
+        if (pluginFilter && !pluginFilter.includes(plugin.manifest.id)) {
+          return;
+        }
+
         const id = plugin.manifest.id;
         const pluginApi = pluginApiMap[id];
-        const info = {
-          id,
-          apiCount: countApiForPlugin(pluginApi),
-          apiCountMissingComments: countMissingCommentsApiForPlugin(pluginApi),
-          missingApiItems: missingApiItems[id],
-        };
-
-        if (info.apiCount > 0) {
+        const apiCount = countApiForPlugin(pluginApi);
+        const pluginStats = collectApiStatsForPlugin(pluginApi);
+
+        reporter.metrics([
+          {
+            id,
+            group: 'API count',
+            value: apiCount,
+          },
+          {
+            id,
+            group: 'API count missing comments',
```

Comment:
If you'd like these to render more prominently on PRs (like below) I can update the bot with descriptions of each group including links to docs or descriptions about how to interpret the changes

![image](https://user-images.githubusercontent.com/1329312/113652447-a67db500-9648-11eb-8459-9a8662259bb1.png)

Reply:
Awesome! I will work on adding doc links. For the time being,:


> Any counts in public APIs
> Total count of every `any` typed public API. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats any` for more detailed information
> 


> Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.

> Public APIs missing exports
> Total count of every type that is part of your API that _should_ be exported but is not. This will cause broken links in the API documentation system. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats exports` for more detailed information.

^^ Something like above??

Reply:
LGTM, @elasticmachine merge upstream so we can see the results